Enigmatry.Entry.Blueprint.Infrastructure/Api/Init/SerilogStartupExtension.cs

```diff
@@ -4,34 +4,40 @@
 using Serilog;
 using Serilog.Sinks.ApplicationInsights.TelemetryConverters;
 using System.Reflection;
-using Microsoft.AspNetCore.Builder;
+using Enigmatry.Entry.Core.Helpers;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.Extensions.DependencyInjection;
 
 namespace Enigmatry.Entry.Blueprint.Infrastructure.Api.Init;
 
 public static class SerilogStartupExtension
 {
-    public static void AppConfigureSerilog(this WebApplicationBuilder builder)
+    public static LoggerConfiguration AppConfigureSerilog(this LoggerConfiguration loggerConfiguration, IConfiguration configuration)
     {
-        LoggerConfiguration loggerConfiguration = new LoggerConfiguration()
-            .ReadFrom.Configuration(builder.Configuration)
+        loggerConfiguration
+            .ReadFrom.Configuration(configuration)
             .Enrich.FromLogContext()
             .Enrich.WithThreadId()
             .Enrich.WithProcessId()
             .Enrich.WithMachineName()
             .Enrich.With(new OperationIdEnricher())
             .Enrich.WithProperty("AppVersion", Assembly.GetEntryAssembly()!.GetName().Version!);
 
-        AddAppInsightsToSerilog(loggerConfiguration, builder.Configuration);
-
-        Log.Logger = loggerConfiguration.CreateLogger();
-
         // for enabling self diagnostics see https://github.com/serilog/serilog/wiki/Debugging-and-Diagnostics
         // Serilog.Debugging.SelfLog.Enable(Console.Error);
+
+        return loggerConfiguration;
     }
 
-    private static void AddAppInsightsToSerilog(LoggerConfiguration loggerConfiguration, IConfiguration configuration)
+    public static void AddAppInsightsToSerilog(this LoggerConfiguration loggerConfiguration, IConfiguration configuration, IServiceProvider serviceProvider)
     {
         var settings = configuration.ReadApplicationInsightsSettings();
-        loggerConfiguration.WriteTo.ApplicationInsights(new TraceTelemetryConverter(), settings.SerilogLogsRestrictedToMinimumLevel);
+        if (settings == null || !settings.ConnectionString.HasContent())
+        {
+            return;
+        }
+
+        var telemetryConfiguration = serviceProvider.GetRequiredService<TelemetryConfiguration>();
+        loggerConfiguration.WriteTo.ApplicationInsights(telemetryConfiguration, TelemetryConverter.Traces);
```

Comment:
In previous implementation we used settings.SerilogLogsRestrictedToMinimumLevel. It's missing here.

Reply:
I wonder if we should configure the application insights sink via the configuration (appsettings.json) as well.
https://github.com/serilog-contrib/serilog-sinks-applicationinsights?tab=readme-ov-file#configuring-with-readfromconfiguration

We get out of the box support to override the minimum level for the sink.

We get rid the both AddAppInsightsToSerilog() extension method and probably also ReadApplicationInsightsSettings() method?

Only drawback that I see is that it's needed to specify the app insights connection string twice: in the app insights settings section and in the app insight serilog sink section. Connection string is replaced by deployment, so it's not a real drawback. (it's not like someone is going to forget to replace it on both places)